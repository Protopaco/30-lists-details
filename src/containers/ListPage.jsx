import React, { useState, useEffect } from 'react';
import { fetchCharacters } from '../../services/fetches.js';
import styles from './ListPage.jsx';
import CharacterPreview from '../components/CharacterPreview/CharacterPreview.jsx'

const ListPage = () => {
    const [characterList, setCharacterList] = useState([]);

    useEffect(async () => {
        const characterList = await fetchCharacters();
        setCharacterList(characterList);
    }, [])

    const characterMap = characterList.map((character, index) => {
        return <li key={index}>
            <CharacterPreview {...character} />
        </li>
    })

    return (
        <div>
            {characterList ?
                <ul className={styles.listPage}>
                    {characterMap}
                </ul>
                :
                <p>LOADING!</p>
            }
        </div>
    )
}

export default ListPage;
