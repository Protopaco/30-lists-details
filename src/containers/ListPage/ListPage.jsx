import React, { useState, useEffect } from 'react';
import { fetchCharacters } from '../../services/fetch.js';
import styles from './ListPage.jsx';
import CharacterPreview from '../../components/CharacterPreview/CharacterPreview.jsx'

const ListPage = () => {
    const [characterList, setCharacterList] = useState([]);
    const [loading, setLoading] = useState(true)

    useEffect(async () => {
        const fetchedList = await fetchCharacters();
        setCharacterList(fetchedList);
        setLoading(false)
    }, [])

    const characterMap = characterList.map((character, index) => {
        if (character.photoUrl) {
            return <li key={index}>
                <CharacterPreview {...character} />
            </li>
        }
        else {
            return <p key={index}></p>
        }
    })

    if (loading) return <p>LOADING!</p>;

    return <ul className={styles.listPage}
        data-testid='characterList'>
        {characterMap}
    </ul>

}

export default ListPage;
