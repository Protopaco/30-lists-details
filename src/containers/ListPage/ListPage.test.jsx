import React from 'react';
// import reactDOM from 'react-dom';
import { act, render, screen, waitFor, cleanup } from '@testing-library/react';
import ListPage from './ListPage';
import { BrowserRouter as Router } from 'react-router-dom';

describe('ListPage tests', () => {
    afterEach(() => cleanup());

    it('displays ListPage', async () => {
        let ul;
        render(
            <Router>
                <ListPage />
            </Router>)

        screen.getByText('LOADING!');
        ul = await screen.findByTestId('characterList');

        return waitFor(() => {
            expect(ul).not.toBeEmptyDOMElement();
        })
    })


    it('renders ListPage', () => {
        const { asFragment } = render(<Router>
            <ListPage />
        </Router>);
        expect(asFragment()).toMatchSnapshot();
    });
})