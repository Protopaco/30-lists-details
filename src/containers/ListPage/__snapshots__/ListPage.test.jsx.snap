// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage tests renders ListPage 1`] = `
<DocumentFragment>
  <p>
    LOADING!
  </p>
</DocumentFragment>
`;
